app/uk/gov/hmrc/gform/sharedmodel/TaxPeriods.scala

```diff
@@ -70,9 +74,32 @@ object TaxPeriodInformation {
 
 sealed trait Obligations
 final case object NotChecked extends Obligations
-final case class RetrievedObligations(listOfObligations: NonEmptyList[TaxPeriodInformation]) extends Obligations
+final case class RetrievedObligations(listOfObligations: List[TaxPeriodInformation]) extends Obligations
 
 object Obligations {
-  import JsonUtils._
-  implicit val format: OFormat[Obligations] = derived.oformat
+  implicit val format: OFormat[Obligations] = derived.oformat[Obligations]
+}
+
+case class IdNumberValue(value: String) extends AnyVal
+
+object IdNumberValue {
+  implicit val format: OFormat[IdNumberValue] = derived.oformat
+}
+
+case class HmrcTaxPeriodWithEvaluatedId(hmrcTaxPeriod: HmrcTaxPeriod, idNumberValue: IdNumberValue)
+
+object HmrcTaxPeriodWithEvaluatedId {
+
+  implicit def readsNonEmptyList[T: Reads] = Reads[NonEmptyList[T]] { json =>
+    Json.fromJson[List[T]](json).flatMap {
+      case Nil     => JsError(ValidationError(s"Required at least one element. Got: $json"))
+      case x :: xs => JsSuccess(NonEmptyList(x, xs))
+    }
+  }
+
+  implicit def writesNonEmptyList[T: Writes] = Writes[NonEmptyList[T]] { v =>
+    JsArray((v.head :: v.tail).map(Json.toJson(_)).toList)
```

Comment:
v.head and v.tail  can we get a NullPointerException to these if v.size == 0 ?

Reply:
Not sure. This is for writing a NonEmptyList though so size should never be 0.
Also, this code is almost directly copied from a different reads and writes so I imagine if it's fine there it should also be here